src/main/java/com/cloudbees/plugins/deployer/engines/Engine.java

```diff
@@ -205,6 +206,33 @@ public DeployedApplicationLocation process(File applicationFile, T target) throw
         }
     }
 
+    private DeployedApplicationLocation addToFacects(AbstractMap.SimpleEntry<String, DeployedApplicationLocation> pair) throws IOException {
+        DeployedApplicationLocation location = pair.getValue();
+        String md5sum = pair.getKey();
+
+        if (!NO_MD5.equals(md5sum)) {
+            Jenkins j = Jenkins.getInstance();
+            if (j == null) {
```

Comment:
can use `getActiveInstance` in suitable core versions (later it is back to `getInstance`)